lib/matplotlib/table.py

```diff
@@ -146,6 +147,57 @@ def set_text_props(self, **kwargs):
         self._text.update(kwargs)
 
 
+class CustomCell(Cell):
+    """
+    A subclass of Cell where the sides may be visibly toggled.
+
+    """
+
+    EDGES = 'BRTL'
+    EDGE_ALIASES = {'open':         '',
+                    'closed':       EDGES,  # default
+                    'horizontal':   'BT',
+                    'vertical':     'RL'
+                    }
+
+    def __init__(self, *args, **kwargs):
+
+        visible_edges = kwargs.pop('visible_edges')
+        Cell.__init__(self, *args, **kwargs)
+
+        if visible_edges is None:
+            self._visible_edges = self.EDGES
+        elif visible_edges in self.EDGE_ALIASES:
+            self._visible_edges = self.EDGE_ALIASES[visible_edges]
+        else:
+            for edge in visible_edges:
+                if edge not in self.EDGES:
+                    msg = 'Invalid edge param {0}, must only be one of {1} or \
```

Comment:
`("something {0}", "something {1}").format(...)` doesn't work as it's a tuple.

`ValueError("something", ..., "something", ...)` seems to just raise it as `ValueError: ("something", ..., "something", ...)` and doesn't implicitly concatenate the strings.